tavern/app.go

```diff
@@ -107,54 +111,73 @@ func NewServer(ctx context.Context, options ...func(*Config)) (*Server, error) {
 		createTestData(ctx, client)
 	}
 
-	// Setup HTTP Handlers
+	// Configure Authentication
+	var withAuthentication tavernhttp.Option
+	if cfg.oauth.ClientID != "" {
+		withAuthentication = tavernhttp.WithAuthenticationCookie(client)
+	} else {
+		withAuthentication = tavernhttp.WithAuthenticationBypass(client)
+	}
+
+	// Configure Request Logging
 	httpLogger := log.New(os.Stderr, "[HTTP] ", log.Flags())
-	router := http.NewServeMux()
-	router.Handle("/status", newStatusHandler())
-	router.Handle("/oauth/login", auth.NewOAuthLoginHandler(cfg.oauth, privKey))
-	router.Handle("/oauth/authorize", auth.NewOAuthAuthorizationHandler(cfg.oauth, pubKey, client, "https://www.googleapis.com/oauth2/v3/userinfo"))
-	router.Handle("/graphql", newGraphQLHandler(client))
-	router.Handle("/cdn/", cdn.NewDownloadHandler(client))
-	router.Handle("/cdn/upload", cdn.NewUploadHandler(client))
-	router.Handle("/", auth.WithLoginRedirect("/oauth/login", www.NewHandler(httpLogger)))
-	router.Handle("/playground", auth.WithLoginRedirect("/oauth/login", playground.Handler("Tavern", "/graphql")))
+
+	// Route Map
+	routes := tavernhttp.RouteMap{
+		"/status":      tavernhttp.Endpoint{Handler: newStatusHandler()},
+		"/oauth/login": tavernhttp.Endpoint{Handler: auth.NewOAuthLoginHandler(cfg.oauth, privKey)},
+		"/oauth/authorize": tavernhttp.Endpoint{Handler: auth.NewOAuthAuthorizationHandler(
+			cfg.oauth,
+			pubKey,
+			client,
+			"https://www.googleapis.com/oauth2/v3/userinfo",
+		)},
+		"/graphql":    tavernhttp.Endpoint{Handler: newGraphQLHandler(client)},
+		"/c2.C2/":     tavernhttp.Endpoint{Handler: newGRPCHandler(client)},
+		"/grpc/":      tavernhttp.Endpoint{Handler: newGRPCHandler(client)},
```

Comment:
Are both of these endpoints needed?

Reply:
Yes, because eventually when the bug is fixed the requests will be sent to /grpc/, but we need to keep /c2.C2 in the meantime.

Reply:
Is `/grpc/` used in the current (broken) iteration?

Reply:
It will be as soon as the bug is fixed